src/main/java/com/openlattice/chronicle/services/message/MessageService.kt

```diff
@@ -0,0 +1,105 @@
+package com.openlattice.chronicle.services.message
+
+import com.google.common.collect.ArrayListMultimap
+import com.google.common.collect.ImmutableSet
+import com.google.common.collect.ListMultimap
+import com.openlattice.chronicle.constants.EdmConstants
+import com.openlattice.chronicle.constants.MessageDetails
+import com.openlattice.chronicle.constants.MessageOutcome
+import com.openlattice.chronicle.services.ApiCacheManager
+import com.openlattice.chronicle.services.edm.EdmCacheManager
+import com.openlattice.chronicle.services.enrollment.EnrollmentManager
+import com.openlattice.chronicle.services.entitysets.EntitySetIdsManager
+import com.openlattice.chronicle.services.twilio.TwilioManager
+import com.openlattice.chronicle.util.ChronicleServerUtil
+import com.openlattice.client.ApiClient
+import com.openlattice.data.DataAssociation
+import com.openlattice.data.DataGraph
+import org.slf4j.LoggerFactory
+import java.time.OffsetDateTime
+import java.util.*
+import java.util.concurrent.ExecutionException
+
+/**
+ * @author toddbergman &lt;todd@openlattice.com&gt;
+ */
+class MessageService(
+    private val apiCacheManager: ApiCacheManager,
+    private val edmCacheManager: EdmCacheManager,
+    private val enrollmentManager: EnrollmentManager,
+    private val entitySetIdsManager: EntitySetIdsManager,
+    private val twilioManager: TwilioManager
+) : MessageManager {
+    override fun sendMessage(
+        organizationId: UUID,
+        studyId: UUID,
+        participantId: String,
+        messageDetails: Map<String, String>
+    ) {
+        val message = MessageDetails(
+            messageDetails?.get("type") ?: "",
+            OffsetDateTime.now(),
+            participantId,
+            messageDetails?.get("phone") ?: "",
+            messageDetails?.get("url") ?: ""
+        )
```

Comment:
Shouldn't have to do this. Looks like it's easier to make messageDetails be of type MessageDetails instead of Map<String, String>

Reply:
should use getValue to enforce that keys should exist

Reply:
should I construct the MessageDetails in ChronicleControllerV2?

Reply:
If you make the change from my comment, you shouldn't have to do anything extra because `MessageDetails` will be passed in the controller.